### ISSUE TITLE
Repetitive output from fine-tuned model

### ISSUE THREAD

**Issue by arjungpt28** [bytedance/video-SALMONN-2#20]
Hi @TCL606 @DragonAura,

I'm in the process of fine-tuning your final model via SFT, and I'm noticing something a bit strange. While the train and val loss go down smoothly, the predictions from my final fine-tuned model on the val set seem very repetitive (either repeating the sentence, or `<im_start>`, or both):

```
{
        "id": [
            "videos/cb8df5e7-af5c-4211-b779-0568f94c7ded.mp4",
            "audio/cb8df5e7-af5c-4211-b779-0568f94c7ded.wav"
        ],
        "prompt": [
            {
                "from": "human",
                "value": "<image>\nWhat was the main function of the item in my lap?"
            }
        ],
        "pred": "It was a bowl of cotton.\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|> a bowl of cotton.\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|> a cup.\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|> a remote control.\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\n<|im_start|>\nYou are seated in a cozy living room.\n<|im_start|>\n<|im_start|>\n<|im_start|>\nYou are eating from a bowl of cotton.\n<|im_start|>\nYou are watching television.\nYou are wearing patterned pants.\nYou are holding a remote control.\nYou are seated on a chair.\nYou are engaged in a conversation.\nYou are interacting with a dog.\nYou are sitting on the floor.\nYou are standing near the door.\nYou are seated on a sofa.\nYou are seated at a table.\nYou are eating.\nYou are knitting.\nYou are engaged in a phone call.\nYou are holding a cup.\nYou are holding a bowl.\nYou are holding a spoon.\nYou are holding a pen.\nYou are holding a pen and a cup.\nYou are holding a pen and a spoon.\nYou are holding a pen and a bowl.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a bowl.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen and a spoon.\nYou are holding a pen",
        "ref": "<|im_start|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|im_end|>\n<|im_start|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|im_end|>\n<|im_start|><|endoftext|><|endoftext|>"
    },
```

In trying to debug this, I have confirmed that:

1. The input and ground truth coming from the data loader and going into the loss function are correct after decoding them (no repetition):

```
[MODEL INPUT -- input_ids]
"<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\n<|endoftext|>\n\nWhat prompted the transition in the room's wall decorations from one event to the next?<|im_end|>\n<|im_start|>assistant\nThe transitions were likely driven by thematic considerations, aiming to reflect evolving visual stories and contextual mood, changing from plain walls to intricate wooden installations and framed art to complement the room's diverse celebrations and gatherings.<|im_end|>\n"


[GROUND TRUTH -- labels]
"<|im_start|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|im_end|>\n<|im_start|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|endoftext|><|im_end|>\n<|im_start|><|endoftext|><|endoftext|>The transitions were likely driven by thematic considerations, aiming to reflect evolving visual stories and contextual mood, changing from plain walls to intricate wooden installations and framed art to complement the room's diverse celebrations and gatherings.<|im_end|>\n"
```

2. When I use the inference code to evaluate your final model, I observe no repetitions (suggesting that the inference code / settings seem to be working as expected).

3. During inference, the `<im_end>` token is never produced by my fine-tuned model, so it never triggers the stopping condition. 

I understand that I haven't provided much information, but I was wondering if you have seen anything similar in your vast experience training such models, or if you can give me any advice to debug this? Any ideas will be much appreciated, thanks so much!
